feat: implement hero section with interactive banner and 3D flip animation

thomaslnx opened the pull request:
- Add responsive hero component with grid background and grain overlay
- Implement 3D flip animation for the main heading with dual messaging
- Add social media navigation with high-DPI social icons
- Update typography system with .text-headline-back for compliance between the sizes of the two flip words
- Add comprehensive test coverage for the HeaderBanner component
- Remove unused Avatar component and clean up component architecture

monkishtypist commented on apps/web/eslint.config.mjs:
```diff
@@ -9,6 +9,13 @@ const compat = new FlatCompat({
+  ...compat.config({
+    extends: ['next/core-web-vitals', 'next/typescript', 'prettier'],
+    rules: {
+      '@typescript-eslint/no-explicit-any': 'off',
```
Set this to `warn` instead

monkishtypist commented on apps/web/public/hero/Instagram.png:
Icons should be stored as SVG so they can more easily scale.

monkishtypist commented on apps/web/src/app/_styles/globals.css:
```diff
@@ -1,33 +1,69 @@
+  --primary-orange: #ff8a50;
+
+  /* Typography */
+  --sharp-sans: 'Sharp Sans', sans-serif;
```
should we move this to Typography stylesheet?

thomaslnx replied:
Good idea. Keep it clean and in one place only.

monkishtypist commented on apps/web/src/app/_styles/typography.css:
```diff
@@ -26,13 +26,13 @@ html {
 @utility text-9xl-custom {
   @apply font-sharp font-extrabold;
-  font-size: 7.875rem;   /* 126px */
+  font-size: clamp(4rem, 8.5vw, 7.875rem); /* 126px */
```
Revert the clamp. We do not want to clamp anything right now.

monkishtypist commented on apps/web/src/app/_styles/typography.css:
```diff
@@ -152,6 +152,15 @@ html {
     }
   }
 
+  /* "#OpenWhatYouWant" hero */
+  .text-headline-back {
+    @apply text-sharp-33;
+    @variant landscape {
+      font-size: clamp(4rem, 8.5vw, 7.15rem); /* 114.4px */
+      line-height: 7.9375rem; /* 127px */
+    }
+  }
+
```
These values look off from design. Also, do not clamp.

```suggestion
  /* "#OpenWhatYouWant" hero */
  .text-headline-back {
    font-size: 1.8938125rem; /* 30.301px */
    line-height: 1.5673125rem; /* 25.077px */
    @variant landscape {
      font-size: 7.25rem; /* 116px */
      line-height: 6rem; /* 96px */
    }
  }

```

Also, these component classes are a perfect place to add text-specific styling like gradient text clipping if you want to keep it cleaner.

monkishtypist commented on apps/web/src/components/Container.tsx:
```diff
@@ -0,0 +1,12 @@
+  children: ReactNode;
+}
+
+const Container = ({ children }: ContainerProps) => {
+  return <div className="flex flex-col w-full max-w-7xl mx-auto">{children}</div>;
+};
```
Let's set a custom css variable to control the container width called `--container-max-width`

Those widths should be:
- Portrait (default): 22.375rem /* 358px */
- Landscape: 82.0625rem /* 1312px */

```suggestion
const Container = ({ children }: ContainerProps) => {
  return <div className="flex flex-col w-full max-w-(--container-max-width) mx-auto">{children}</div>;
};
```

monkishtypist commented on apps/web/src/components/HeaderBanner/index.tsx:
```diff
@@ -0,0 +1,137 @@
+  return (
+    <div
+      className="hero-bbo-header flex flex-col relative items-center justify-center w-full h-[16.25rem]
+                 rounded-[12px] border-[var(--colors-common-ash)] border-[2px] bg-[var(--primary-charcoal)]
```
convert all `px` to `rem`

monkishtypist commented on apps/web/src/components/HeaderBanner/index.tsx:
```diff
@@ -0,0 +1,137 @@
+              className="text-headline-back bg-gradient-to-r from-[var(--secondary-yellow)] to-[var(--primary-orange)]
+                         bg-clip-text text-transparent"
+            >
+              <h2 className="flip-heading" aria-label="Alternative heading: Open What You Want">
```
change this `h2` to a `span` to avoid confusing SEO hierarchy.

monkishtypist commented on apps/web/src/components/HeaderBanner/index.tsx:
```diff
@@ -0,0 +1,137 @@
+      </div>
+
+      <span className="powered-by relative flex w-full justify-end items-center">
+        <p className="mr-[8px] font-[family-name:var(--sharp-sans)] text-[10px] font-medium tracking-[0.8] z-20">
```
convert all `px` to `rem`

Also it might make sense to have the word "Firefox" in the paragraph text but invisible (like font-size: 0.001, opacity: 0) for SEO purposes...

monkishtypist commented on apps/web/src/components/HeaderBanner/index.tsx:
```diff
@@ -0,0 +1,137 @@
+          ></Image>
+        </Link>
+      </span>
+      <div className="header-bottom flex w-full h-[2.9375rem] px-[0.5rem] py-0 justify-between items-center mt-[18px]">
```
convert px to rem

monkishtypist commented on apps/web/src/components/HeaderBanner/index.tsx:
```diff
@@ -0,0 +1,137 @@
+      </span>
+      <div className="header-bottom flex w-full h-[2.9375rem] px-[0.5rem] py-0 justify-between items-center mt-[18px]">
+        <nav
+          className="left flex w-[26.75rem] shrink-0 items-start gap-[2rem] text-[12px] uppercase font-bold
```
I believe the `.label-14` class is for nav text

monkishtypist commented on apps/web/src/components/HeaderBanner/index.tsx:
```diff
@@ -0,0 +1,137 @@
+          </Link>
+        </nav>
+        <div className="right flex items center">
+          <nav className="flex gap-[1rem] items-center justify-center">
```
Pull the params out into an object so you can iterate over that. Something like:

```
[
  {
    title: "Twitch",
    href: "https://www.twitch.tv/",
    "aria-label": "Visit our Twitch channel"
  },
  { ... }
]
```

thomaslnx replied:
For now, the social icons were removed till the final layout decision.

monkishtypist commented on apps/web/src/components/Hero/index.tsx:
Do we need this component anymore? The HeaderBanner is now stand-alone, so we should be able to remove this.

thomaslnx replied:
Actually, I put the Hero component as the entire microsite wrapper. Indeed, the grid and grain background images are styled inside it. But if you think is better, I can put all those inside the Container component.

monkishtypist commented on apps/web/src/components/Layout.tsx:
Where is this called? Why are we using a component-level Layout?

monkishtypist reviewed (changes requested):
@thomaslnx I have reviewed and added some comments and requested a few changes. Overall, remember to never use any explicit `px` values - always convert to `rem` using formula `px` / 16 = `rem`.

thomaslnx commented:
## Overview
Refactors the hero section to improve maintainability and performance by consolidating assets and simplifying the HeaderBanner component architecture.

## Changes
- **Asset Migration**: Consolidated hero assets from `/hero/` to `/assets/images/` with SVG optimization
- **Component Simplification**: Replaced complex HeaderBanner with minimal NightSky background implementation
- **Performance**: Removed 30+ unused PNG assets, reducing bundle size
- **Code Quality**: Eliminated legacy test files and unused components
- **Responsive Design**: Enhanced typography utilities with CSS custom properties for better mobile/desktop scaling

## Technical Details
- Migrated from PNG to SVG assets for better scalability and performance
- Implemented responsive typography system with viewport-based calculations
- Streamlined component structure, removing 135+ lines of complex JSX
- Updated asset paths throughout Hero component hierarchy

## Testing
- [x] Responsive breakpoints verified
- [x] Asset loading confirmed across devices

## Related
Closes #10

thomaslnx commented on apps/web/src/app/_styles/typography.css:
```diff
@@ -152,6 +184,15 @@ html {
     }
   }
 
+  /* "#OpenWhatYouWant" hero */
+  .text-headline-back {
+    @apply text-sharp-33;
+    @variant landscape {
+      font-size: clamp(4rem, 8.5vw, 7.15rem); /* 114.4px */
+      line-height: 7.9375rem; /* 127px */
+    }
+  }
+
```
done.

thomaslnx commented on apps/web/src/app/_styles/typography.css:
```diff
@@ -8,9 +8,36 @@
+  /* Typography */
+  --sharp-sans: 'Sharp Sans', sans-serif;
+
+  /* main-logo */
+  --logo-width: 25.983rem;
+  --logo-height: 12.463rem;
+
```
done.

thomaslnx commented on apps/web/src/components/HeaderBanner/index.tsx:
```diff
@@ -0,0 +1,12 @@
+  return (
+    <>
+      <div
+        className="hero-bbo-header header-banner-height flex flex-col relative items-center justify-center w-full
```
Done.

thomaslnx commented on apps/web/src/components/Hero/index.tsx:
Done.

monkishtypist commented on apps/web/src/app/_styles/typography.css:
```diff
@@ -215,4 +256,17 @@ html {
     }
   }
+
+  .hero-bbo-header {
+    height: var(--header-banner-height-portrait);
+    margin-top: var(--header-banner-margin-top-portrait);
+    border-width: var(--header-banner-border-width-portrait);
+    border-radius: var(--header-banner-border-radius-portrait);
+  }
+
+  .header-banner-height {
+    @variant landscape {
+      @apply landscape-header;
+    }
+  }
```
Maybe create a "Card" component that just wraps each block with border radius and width and color props, plus a background image prop.

monkishtypist replied:
```
<Card props={props}>
   {children}
</Card>
```

monkishtypist commented on apps/web/src/components/Header/index.tsx:
```diff
@@ -0,0 +1,14 @@
+const Header = () => {
+  return (
+    <>
+      <div
+        className="flex flex-col h-16 landscape:h-[10.9375rem] portrait:h-[2.98rem] relative
+                   items-center justify-center w-full border-[var(--colors-common-ash)] rounded-[0.75rem]
+                   portrait:rounded-[0.3rem] border-[0.125rem] portrait:border-[0.035rem] portrait:mt-[1.15rem] landscape:mt-[4rem]
+                   bg-no-repeat bg-cover bg-[url(/assets/images/NightSky.svg)]"
+      ></div>
```
You do not need to add `portrait:` variant because the default styles are already portrait. And you only need to add the `landscape:` variant if it is different from the default.

```suggestion
      <div
        className="flex flex-col h-16 landscape:h-[10.9375rem] relative
                   items-center justify-center w-full border-[var(--colors-common-ash)] rounded-[0.75rem]
                   border-[0.125rem] landscape:mt-[4rem]
                   bg-no-repeat bg-cover bg-[url(/assets/images/NightSky.svg)]"
      ></div>
```
